### PR TITLE
Feature card list screen

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
PR created to merge the Pokemon Card List feature with master.

Tests of the feature with coverage:

test passed cardList database

test coverage cardList database

Sthetho debugging the database:

Stetho debugging database

GIF of the app:

card list database

### CODE REVIEW

File: app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt [hunk 1]
```diff
@@ -51,21 +53,34 @@ class PokemonTypeFragment : Fragment() {
...
+        Toast.makeText(context, error, Toast.LENGTH_SHORT).show()
     }
 
     companion object {
+        private const val TYPE = "types"
         private const val COLUMNS_PORTRAIT = 4
         private const val COLUMNS_LANDSCAPE = 6
     }
+
+    override fun onPokemonTypeSelected(typeSelected: PokemonType) {
```
Comment (SartiMau) [inline app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt]: this should call the viewModel to set the new state of this liveData.. then we will call the startActivity

File: app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt [hunk 2]
```diff
@@ -51,21 +53,34 @@ class PokemonTypeFragment : Fragment() {
...
         }
     }
 
     private fun showPokemonTypesError(error: String?) {
         binding.pokemonAlltypesLoading.visibility = View.GONE
-        error?.let { Toast.makeText(context, it, Toast.LENGTH_SHORT).show() }
+        Toast.makeText(context, error, Toast.LENGTH_SHORT).show()
     }
 
     companion object {
```
Comment (SartiMau) [inline app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt]: please put the companion at the end of the class

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: data/src/main/java/com/globant/data/database/PokemonCardDatabaseImpl.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+class PokemonCardDatabaseImpl(private val pokemonDao: PokemonDao) : PokemonCardDatabase {
+    private val pokemonCardDatabaseMapper = PokemonCardDatabaseMapper()
+
+    override fun getLocalPokemonCardList(pokemonCardGroup: String, pokemonCardGroupSelected: String): Result<List<PokemonCard>> {
+        val pokemonCardList: List<PokemonCardDatabaseEntity> =
+            when (pokemonCardGroup) {
+                TYPE -> pokemonDao.getPokemonCardListType(pokemonCardGroupSelected)
+                SUPERTYPE -> pokemonDao.getPokemonCardListSupertype(pokemonCardGroupSelected)
+                SUBTYPE -> pokemonDao.getPokemonCardListSubtype(pokemonCardGroupSelected)
+                else -> listOf()
```
Comment (lucianoeli) [inline data/src/main/java/com/globant/data/database/PokemonCardDatabaseImpl.kt]: Here you should send an error, explaining there is not such kind of  card group

File: app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt (hunk 2 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt]: done, thanks

File: app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt (hunk 1 above)
Comment (MarcosCGlobant) [reply app/src/main/java/com/globant/pokemontcgapp/fragment/PokemonTypeFragment.kt]: refactored and updated the unit tests to check the functionality, thanks for the explanation

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**